### ISSUE TITLE
apoc.refactor.mergeNodes() has problems coping with uniqueness constraints - Reopening

### ISSUE THREAD

**Issue by jvhashe** [neo4j-contrib/neo4j-apoc-procedures#1175]
I'm experiencing the exact same issue described here.

https://github.com/neo4j-contrib/neo4j-apoc-procedures/issues/762

The original issue was root-caused to be a problem in Neo4j. After seeing this regression, I contacted Neo4j customer support. They claimed it's an issue in APOC. Their response is below.

  - OS: CentOS 7 / Alpine Linux
  - Neo4j: 3.4.X and 3.5.X
  - Neo4j-Apoc: 3.4.X and 3.5.X

> This is an APOC bug because it tries to write the source properties on the target node before the target node is deleted: 
>https://github.com/neo4j-contrib/neo4j-apoc-procedures/blob/3f74d1c37a024195e373b001ae353d77c402776e/src/main/java/apoc/refactor/GraphRefactoring.java#L481-L499 
>If they first copied the source node properties, then deleted the source node, and then write the properties to the target node, then it would work. The test that is added in his original PR proves this:
>https://github.com/neo4j/neo4j/blob/3.5/community/community-it/index-it/src/test/java/org/neo4j/kernel/api/impl/index/AccidentalUniquenessConstraintViolationIT.java

**Comment by conker84** [comment]
@jvhashe are you already experiencing the issue?
I tried with the following use-case:
```
CREATE CONSTRAINT ON (a:A) ASSERT a.prop1 IS UNIQUE;
CREATE CONSTRAINT ON (a:B) ASSERT a.prop2 IS UNIQUE;
CREATE (a:A) SET a.prop1 = 1;
CREATE (b:B) SET b.prop2 = 99;

MATCH (a:A {prop1:1})
MATCH (b:B {prop2:99})
CALL apoc.refactor.mergeNodes([a, b]) YIELD node RETURN node;
```

And it worked.
Can detail your use-case?

**Comment by jvhashe** [comment]
@conker84 This has now been fixed in APOC version 3.5.0.4. I'll close the bug.
